optimizely/helpers/condition.py

```diff
@@ -233,12 +238,74 @@ def substring_evaluator(self, index):
 
         return condition_value in user_value
 
+    def semver_equal_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) == 0
+
+    def semver_greater_than_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) == 1
+
+    def semver_less_than_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) == -1
+
+    def semver_less_than_or_equal_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) <= 0
+
+    def semver_greater_than_or_equal_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) >= 0
+
+    def compare_user_version_with_target_version(self, index):
+
+        condition_name = self.condition_data[index][0]
+        target_version = self.condition_data[index][1]
+        user_version = self.attributes.get(condition_name)
+
+        if not isinstance(target_version, string_types):
+            self.logger.warning(audience_logs.UNKNOWN_CONDITION_VALUE.format(self._get_condition_json(index),))
+            return None
+
+        if not isinstance(user_version, string_types):
+            self.logger.warning(
+                audience_logs.UNEXPECTED_TYPE.format(self._get_condition_json(index), type(user_version), condition_name)
+            )
+            return None
+
+        target_version_parts = target_version.split(".")
+        user_version_parts = user_version.split(".")
+        user_version_parts_len = len(user_version_parts)
+
+        for (idx, _) in enumerate(target_version_parts):
+            if user_version_parts_len <= idx:
+                return -1
+            # compare strings e.g: n1.n2.n3-alpha/beta
+            if not user_version_parts[idx].isdigit():
+                if user_version_parts[idx] != target_version_parts[idx]:
```

Comment:
my understanding here is that if string contains alpha/beta, we look for exact match. If this is the case, we should also exit with 0 here if it matches.

Reply:
We are just copying logic from swift, so no worries. w